lib/id3c/cli/command/manifest.py

```diff
@@ -192,30 +246,54 @@ def parse_using_config(config_file):
             kwargs = {
                 "workbook": config["workbook"],
                 "sheet": config["sheet"],
-                "sample_column": config["sample_column"],
+                "sample_column": config.get("sample_column"),
+                "collection_column": config.get("collection_column"),
+                "date_column": config.get("date_column"),
                 "extra_columns": list(config.get("extra_columns", {}).items()),
-                "sample_type": config.get("sample_type")
+                "sample_type": config.get("sample_type"),
+                "row_filter" : config.get("row_filter")
             }
         except KeyError as key:
             LOG.error(f"Required key «{key}» missing from config {config}")
             raise key from None
 
+        if not config.get("sample_column") and not config.get("collection_column"):
+            raise KeyError("The configuration must indicate the sample column, the collection "
+                + "column, or both.")
+
+        reserved_columns_in_extra_columns = find_reserved_columns_used_in_extra_column_list(kwargs["extra_columns"])
+        assert len(reserved_columns_in_extra_columns) == 0, \
+            f"A reserved column name has been configured in the extra_columns setting: {reserved_columns_in_extra_columns}"
+
```

Comment:
I think it'd be better to put both these two checks and their equivalents in `parse()` above into `_parse()` below instead. Or maybe you considered this but rejected it?

Reply:
I had the same checks in both "outer" functions so that I could give a tailored error message for the sample_column and collection_column part. Every time I looked at this I thought "Yeah, but you should also do the checks in _parse to be safe". Doing it in all three places felt overboard.
I moved these two checks into _parse. That's the right place for them to be.